tests/klay_blockchain_test.go

```diff
@@ -214,88 +217,89 @@ func deployRandomTxs(t *testing.T, txpool work.TxPool, chainId *big.Int, sender
+	require.True(t, err == nil || err == blockchain.ErrAlreadyNonceExistInPool)
+
 	sender.AddNonce()
+	return tx
 }
 
-// deployContractDeployTx deploy contrac
-func deployContractDeployTx(t *testing.T, txpool work.TxPool, chainId *big.Int, sender *TestAccountType, contractDeployCode string) common.Address {
+// deployContractDeployTx deploys a contract
+func deployContractDeployTx(t *testing.T, txpool work.TxPool, chainId *big.Int, sender *TestAccountType, code string) (*types.Transaction, common.Address) {
 	signer := types.LatestSignerForChainID(chainId)
-	gasPrice := big.NewInt(25 * params.Ston)
+	gasPrice := txpool.GasPrice()
 
 	values := map[types.TxValueKeyType]interface{}{
 		types.TxValueKeyNonce:         sender.GetNonce(),
+		types.TxValueKeyFrom:          sender.GetAddr(),
+		types.TxValueKeyTo:            (*common.Address)(nil),
 		types.TxValueKeyAmount:        new(big.Int).SetUint64(0),
-		types.TxValueKeyGasLimit:      uint64(1000000),
+		types.TxValueKeyGasLimit:      uint64(1e9),
 		types.TxValueKeyGasPrice:      gasPrice,
-		types.TxValueKeyHumanReadable: false,
-		types.TxValueKeyFrom:          sender.GetAddr(),
-		types.TxValueKeyData:          common.FromHex(contractDeployCode),
+		types.TxValueKeyData:          common.FromHex(code),
 		types.TxValueKeyCodeFormat:    params.CodeFormatEVM,
-		types.TxValueKeyTo:            (*common.Address)(nil),
+		types.TxValueKeyHumanReadable: false,
 	}
+
 	tx, err := types.NewTransactionWithMap(types.TxTypeSmartContractDeploy, values)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err := tx.SignWithKeys(signer, sender.GetTxKeys()); err != nil {
-		t.Fatal(err)
-	}
-	if err := txpool.AddLocal(tx); err != nil && err != blockchain.ErrAlreadyNonceExistInPool {
-		t.Fatal(err)
-	}
+	require.Nil(t, err)
+
+	err = tx.SignWithKeys(signer, sender.GetTxKeys())
+	require.Nil(t, err)
+
+	err = txpool.AddLocal(tx)
+	require.True(t, err == nil || err == blockchain.ErrAlreadyNonceExistInPool)
+
 	contractAddr := crypto.CreateAddress(sender.Addr, sender.Nonce)
 	sender.AddNonce()
-
-	return contractAddr
+	return tx, contractAddr
 }
 
-func deployContractExecutionTx(t *testing.T, txpool work.TxPool, chainId *big.Int, sender *TestAccountType, contractAddr common.Address) {
+func deployContractExecutionTx(t *testing.T, txpool work.TxPool, chainId *big.Int, sender *TestAccountType, contractAddr common.Address, calldata string) *types.Transaction {
 	signer := types.LatestSignerForChainID(chainId)
-	gasPrice := big.NewInt(25 * params.Ston)
-	contractExecutionPayload := "0x197e70e40000000000000000000000000000000000000000000000000000000000000001"
+	gasPrice := txpool.GasPrice()
 
 	values := map[types.TxValueKeyType]interface{}{
 		types.TxValueKeyNonce:    sender.GetNonce(),
 		types.TxValueKeyFrom:     sender.GetAddr(),
 		types.TxValueKeyTo:       contractAddr,
 		types.TxValueKeyAmount:   new(big.Int).SetUint64(0),
-		types.TxValueKeyGasLimit: uint64(100000),
+		types.TxValueKeyGasLimit: uint64(1e9),
```

Comment:
I think the gas limit would be limited to 1e8. what about `types.TxValueKeyGasLimit: uint64(1e8),` ?

Reply:
Right. It'll be limited to 1e8 as in here: https://github.com/klaytn/klaytn/blob/5f18f2bf80f6fa039b232a359c7d801ef1a5e9e5/params/protocol_params.go#L175 
https://github.com/klaytn/klaytn/blob/5f18f2bf80f6fa039b232a359c7d801ef1a5e9e5/blockchain/vm/interpreter.go#L274

fixed in b0d5a60